exec/java-exec/src/test/java/org/apache/drill/exec/store/parquet/TestParquetMetadataCache.java

```diff
@@ -398,10 +399,115 @@ public void testDrill4877() throws Exception {
 
   }
 
+  @Test // DRILL-4139
+  public void testBooleanPartitionPruning() throws Exception {
+    final String boolPartitionTable = "dfs_test.tmp.`interval_bool_partition`";
+    try {
+      test("create table %s partition by (col_bln) as " +
+            "select * from cp.`parquet/alltypes_required.parquet`", boolPartitionTable);
+      test("refresh table metadata %s", boolPartitionTable);
```

Comment:
We probably want to cover the cases where metadata cache file is not created (run the query before call "refresh table metadata").